From: James McCoy <jamessan@debian.org>
Date: Wed, 1 Aug 2018 20:44:35 -0400
Subject: last-changed-date-charset

Bug #290774: Convert $LastChangedDate$ keyword to the local character
Bug #290774: Convert $LastChangedDate$ keyword to the local character
set, since it is already rendered in the local language.

Not accepted upstream: http://svn.haxx.se/dev/archive-2006-04/0730.shtml
---
 subversion/libsvn_subr/subst.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/subversion/libsvn_subr/subst.c b/subversion/libsvn_subr/subst.c
index c8c3018..a1213cb 100644
--- a/subversion/libsvn_subr/subst.c
+++ b/subversion/libsvn_subr/subst.c
@@ -210,8 +210,12 @@ keyword_printf(const char *fmt,
           break;
         case 'D': /* long format of date of this revision */
           if (date)
-            svn_stringbuf_appendcstr(value,
-                                     svn_time_to_human_cstring(date, pool));
+            {
+              char *date_keyword;
+              char *date_utf8 = svn_time_to_human_cstring(date, pool);
+              SVN_ERR(svn_utf_cstring_from_utf8(&date_keyword, date_utf8, pool));
+              svn_stringbuf_appendcstr(value, date_keyword);
+            }
           break;
         case 'P': /* relative path of this file */
           if (repos_root_url && *repos_root_url != '\0' && url && *url != '\0')
